int8 encoder_a
int8 encoder_b
int8 pwm
int8 fwd_dir
int8 back_dir